libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -800,6 +796,14 @@ void CompilerUtils::convertType(
 			if (targetIntegerType.numBits() < typeOnStack.numBytes() * 8)
 				convertType(IntegerType(typeOnStack.numBytes() * 8), _targetType, _cleanupNeeded);
 		}
+		else if (targetTypeCategory == Type::Category::FixedPoint)
+		{
+			// Conversion from bytes to fixed point type. No need to clean the high bit
+			// only to shift right because of opposite alignment.
+			FixedPointType const& targetFixedPointType = dynamic_cast<FixedPointType const&>(_targetType);
+			rightShiftNumberOnStack(256 - typeOnStack.numBytes() * 8);
+			solAssert(targetFixedPointType.numBits() == typeOnStack.numBytes() * 8, "");
```

Comment:
I'd swap this and the shift, in case the shift would have some other assertion, then it may be harder to debug what went wrong.

Reply:
Yes, will do. Another interesting thing about this code: I copied it from the integer case above. The thing is that it does not do sign extension for signed types. Ok, this is the old codegen which always does lazy cleanup, so I think it's fine, but we should check that it is tested.